### PR TITLE
Statistics/PackageDownloadsDetail throws System.ArgumentNullException #3200

### PR DESCRIPTION
**Pull request by maartenba**
Fixes https://github.com/NuGet/NuGetGallery/issues/3200

Fixed by checking for known dimensions (and if an unknown dimension is
specified, it's ignored).

@xavierdecoster @skofman1 @ryuyu @shishirx34 @scottbommarito

### CODE REVIEW

**Comment by xavierdecoster** [comment]
:shipit:

File: src/NuGetGallery/Controllers/StatisticsController.cs [hunk 1]
```diff
@@ -148,9 +148,10 @@ public StatisticsController(IStatisticsService statisticsService, IAggregateStat
                 return new HttpStatusCodeResult(HttpStatusCode.NotFound);
             }
 
-            StatisticsPackagesReport report = await _statisticsService.GetPackageDownloadsByVersion(id);
+            var dimensions = new [] { "Version", "ClientName", "ClientVersion", "Operation" };
```
Comment (skofman1) [inline src/NuGetGallery/Controllers/StatisticsController.cs]: >Version [](start = 39, length = 7)

consts?

File: src/NuGetGallery/Controllers/StatisticsController.cs [hunk 2]
```diff
@@ -222,7 +225,10 @@ private void ProcessReport(StatisticsPackagesReport report, string[] groupby, st
                     // the pivot array is used as the Columns in the report so we resize because this was the final set of columns
                     Array.Resize(ref pivot, dim);
                 }
+            }
 
+            if (groupby != null)
```
Comment (skofman1) [inline src/NuGetGallery/Controllers/StatisticsController.cs]: >groupby [](start = 16, length = 7)

can we add UTs for this?

File: src/NuGetGallery/Controllers/StatisticsController.cs [hunk 3]
```diff
@@ -204,7 +207,7 @@ private void ProcessReport(StatisticsPackagesReport report, string[] groupby, st
             var pivot = new string[4];
             if (groupby != null)
             {
-                //  process and validate the groupby query. unrecognized fields are ignored. others fields regarded for existance
+                // process and validate the groupby query. unrecognized fields are ignored. others fields regarded for existance
```
Comment (shishirx34) [inline src/NuGetGallery/Controllers/StatisticsController.cs]: nit: existence

**Comment by shishirx34** [comment]
Looks good.

File: src/NuGetGallery/Controllers/StatisticsController.cs (hunk 1 above)
Comment (maartenba) [inline src/NuGetGallery/Controllers/StatisticsController.cs]: Fixed

File: src/NuGetGallery/Controllers/StatisticsController.cs (hunk 2 above)
Comment (maartenba) [inline src/NuGetGallery/Controllers/StatisticsController.cs]: Added a case for this one (in `StatisticsControllerFacts`)

File: src/NuGetGallery/Controllers/StatisticsController.cs (hunk 3 above)
Comment (maartenba) [inline src/NuGetGallery/Controllers/StatisticsController.cs]: Fixed

**Comment by maartenba** [comment]
Did some updates, can you have another pass?

**Comment by xavierdecoster** [comment]
LGTM :ship:

**Comment by shishirx34** [comment]
LGTM :shipit:

**Comment by skofman1** [comment]
:shipit:

**Pull request merged**